Assets/Character/Scripts/PlayerInput.cs

```diff
@@ -2,36 +2,60 @@
 
 public static class PlayerInput
 {
-    public static bool PressedRoll()
+    public static float PressedRoll() //C
     {
-        return Input.GetKeyDown(KeyCode.C);
+        if (Input.GetButtonDown("Roll") && Input.GetAxis("RjoystickX") != 0)
+        {
+            float direction = Input.GetAxis("RjoystickX");
+            return direction;

```

Comment:
I think this return value change is not good. I feel like you should leave PressedRoll() alone, so it only checks for GetButtonDown("Roll"), and use it with the move functions. If you're really concerned with that, make a PressedRollRight()/Left() that wraps it or something.  Either way this was one of the main reasons the new input handling gives your function a set of inputs that frame.